### DISCUSSION TITLE
Skip intro

### DISCUSSION THREAD

**Discussion by FunctionDJ** [jellyfin/jellyfin-androidtv#3793] [General]
Is there any support for any sort of intro skipping (with a button or automatically/forced) when using this Jellyfin client?

**Comment by nielsvanvelzen** [comment]
No, there is no functionality for that in the server

**Reply by FunctionDJ** [reply]
Okay, so i guess that since intro skipping can't simply be implemented with a server plugin, this feature completely depends on JF server implementing it natively, correct?

So i should be looking for a feature request for a native implementation on the JF server, right?

**Reply by nielsvanvelzen** [reply]
Correct. Our official clients only implement functionality exposed by the server API. Sometimes you need plugins to activate it. For skipping segments we do have a proposal for a new "Media Segment API" but unfortunately there hasn't been much progress on it recently.

https://github.com/jellyfin/jellyfin-meta/discussions/30

**Reply by pavichokche** [reply]
I don't get it...there IS an intro-skipping plugin that WORKS with the AndroidTV Jellyfin app, today...it skips credits too. You just enable the "automatically skip intros/credits" option in the plugin's settings. It only doesn't work when using an external player.

**Reply by nielsvanvelzen** [reply]
Well then just use that? Not sure what you're asking for.

**Reply by pavichokche** [reply]
I'm not OP, I'm not asking for anything (you've done/are doing amazing work), I was just very confused by the discussion.

**Comment by johnpc** [comment]
We should re-open this, media segments is available on jellyfin server 10.10.0+

**Reply by nielsvanvelzen** [reply]
No, we're still not adding third-party plugin support.

**Reply by pavichokche** [reply]
@johnpc don't worry about it, it's being implemented into the application already :) you can see it mentioned in the notes for the latest Beta 0.18. Sounds amazing
TECHNICALLY the Android TV application is still not supporting third party plugins - it relies on the segments on the server. However, the server currently needs third-party plugins to provide those segment details. So worry not, everything is good

**Reply by thornbill** [reply]
> However, the server currently needs third-party plugins to provide those segment details.

Not entirely. The chapter segments plugin is first party.

**Reply by pavichokche** [reply]
> > However, the server currently needs third-party plugins to provide those segment details.
> 
> Not entirely. The chapter segments plugin is first party.

Does the origin of the plugin matter? The release post for 10.10 says "For 10.10, we only provide the general structure to store those Media Segments, and you will still require a plugin to create them.". So is this application going to support segments created by the first-party plugin but not segments created by third-party plugins? I sure hope not, that would be very strange...

**Reply by thornbill** [reply]
No there has just been a lot of misinformation being spread about what is required for media segments.
